emqx does not compile in Alpine 3.13

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->
<!-- 请使用英文描述问题 -->

**Environment**:

- EMQ X version (e.g. `emqx_ctl status`): latest
- Hardware configuration (e.g. `lscpu`): alpine 3.13 on virtual machine
- OS (e.g. `cat /etc/os-release`): Alpine 3.13
- Kernel (e.g. `uname -a`):
- Erlang/OTP version (in case you build emqx from source code): Alpine erlang-dev package
- Others:

**What happened and what you expected to happen**:
run make, start compiling, immediatly after 
Compiling emqx_web_hook
===> Compiling emqx_dashboard
===> Compiling emqx_modules
===> Compiling emqx_telemetry
===> [relup_helper] running gen_appups
===> **Uncaught error in rebar_core. Run with DIAGNOSTIC=1 to see stacktrace or consult rebar3.crashdump**
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
make: *** [Makefile:69: emqx] Error 1

**How to reproduce it (as minimally and precisely as possible)**:
cd emqx
make

**Anything else we need to know?**:

Hi,

made make with `export DIAGNOSTIC=1`
 ...

===> Solving Release emqx-4.3-rc.3-2a9a5e74
===> Uncaught error in rebar_core. Run with DIAGNOSTIC=1 to see stacktrace or consult rebar3.crashdump
===> Uncaught error: {badmatch,{error,enoent}}
===> Stack trace to the error location:
[{rlx_resolve,to_app,3,
              [{file,"/src/_build/default/lib/relx/src/rlx_resolve.erl"},
               {line,197}]},
 {rlx_resolve,find_app_in_code_path,2,
              [{file,"/src/_build/default/lib/relx/src/rlx_resolve.erl"},
               {line,180}]},
 {rlx_resolve,subset,6,
              [{file,"/src/_build/default/lib/relx/src/rlx_resolve.erl"},
               {line,66}]},
 {rlx_resolve,solve_release,2,
              [{file,"/src/_build/default/lib/relx/src/rlx_resolve.erl"},
               {line,47}]},
 {relx,build_release_,3,
       [{file,"/src/_build/default/lib/relx/src/relx.erl"},{line,170}]},
 {relx,build_release,3,
       [{file,"/src/_build/default/lib/relx/src/relx.erl"},{line,77}]},
 {rebar_relx,do,2,[{file,"rebar_relx.erl"},{line,67}]},
 {rebar_core,do,2,[{file,"rebar_core.erl"},{line,155}]},
 {rebar_prv_do,do_task,5,[{file,"rebar_prv_do.erl"},{line,75}]},
 {rebar_core,do,2,[{file,"rebar_core.erl"},{line,155}]},
 {rebar_prv_do,do_task,5,[{file,"rebar_prv_do.erl"},{line,75}]},
 {rebar_core,do,2,[{file,"rebar_core.erl"},{line,155}]},
 {rebar3,run_aux,2,[{file,"rebar3.erl"},{line,181}]},
 {rebar3,main,1,[{file,"rebar3.erl"},{line,66}]},
 {escript,run,2,[{file,"escript.erl"},{line,758}]},
 {escript,start,1,[{file,"escript.erl"},{line,277}]},
 {init,start_em,1,[]},
 {init,do_boot,3,[]}]
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
make: *** [Makefile:69: emqx] Error 1




here is the content of  `rebar3.crashdump` 


Error: {badmatch,{error,enoent}}
[{rlx_resolve,to_app,3,
              [{file,"/src/_build/default/lib/relx/src/rlx_resolve.erl"},
               {line,197}]},
 {rlx_resolve,find_app_in_code_path,2,
              [{file,"/src/_build/default/lib/relx/src/rlx_resolve.erl"},
               {line,180}]},
 {rlx_resolve,subset,6,
              [{file,"/src/_build/default/lib/relx/src/rlx_resolve.erl"},
               {line,66}]},
 {rlx_resolve,solve_release,2,
              [{file,"/src/_build/default/lib/relx/src/rlx_resolve.erl"},
               {line,47}]},
 {relx,build_release_,3,
       [{file,"/src/_build/default/lib/relx/src/relx.erl"},{line,170}]},
 {relx,build_release,3,
       [{file,"/src/_build/default/lib/relx/src/relx.erl"},{line,77}]},
 {rebar_relx,do,2,[{file,"rebar_relx.erl"},{line,67}]},
 {rebar_core,do,2,[{file,"rebar_core.erl"},{line,155}]},
 {rebar_prv_do,do_task,5,[{file,"rebar_prv_do.erl"},{line,75}]},
 {rebar_core,do,2,[{file,"rebar_core.erl"},{line,155}]},
 {rebar_prv_do,do_task,5,[{file,"rebar_prv_do.erl"},{line,75}]},
 {rebar_core,do,2,[{file,"rebar_core.erl"},{line,155}]},
 {rebar3,run_aux,2,[{file,"rebar3.erl"},{line,181}]},
 {rebar3,main,1,[{file,"rebar3.erl"},{line,66}]},
 {escript,run,2,[{file,"escript.erl"},{line,758}]},
 {escript,start,1,[{file,"escript.erl"},{line,277}]},
 {init,start_em,1,[]},
 {init,do_boot,3,[]}]

Hi @zmstone ,

===> Solving Release emqx-4.3-rc.3-41168065
escript: exception error: no function clause matching
                 rlx_resolve:format_error({bad_app_file,
                                           "/usr/lib/erlang/lib/observer-2.9.5/ebin/observer.app",
                                           {error,enoent}}) (/src/_build/default/lib/relx/src/rlx_resolve.erl, line 10)
  in function  rebar3:handle_error/2 (rebar3.erl, line 346)
  in call from escript:run/2 (escript.erl, line 758)
  in call from escript:start/1 (escript.erl, line 277)
  in call from init:start_em/1
  in call from init:do_boot/3
make: *** [Makefile:69: emqx] Error 127

--- a/.github/workflows/run_test_cases.yaml
+++ b/.github/workflows/run_test_cases.yaml
@@ -110,8 +110,9 @@ jobs:
             docker exec -i erlang bash -c "make ct"
         - name: run cover
           run: |
+            printenv > .env
             docker exec -i erlang bash -c "make cover"
-            docker exec -i erlang bash -c "make coveralls"
+            docker exec --env-file .env -i erlang bash -c "make coveralls"
         - uses: actions/upload-artifact@v1
           if: failure()
           with:

--- a/.github/workflows/run_test_cases.yaml
+++ b/.github/workflows/run_test_cases.yaml
@@ -122,3 +122,15 @@ jobs:
           with:
             name: cover
             path: _build/test/cover
+
+    finish:
+      needs: run_common_test
+      runs-on: ubuntu-20.04
+      steps:
+        - name: Coveralls Finished
+          env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          run: |
+            curl -v -k https://coveralls.io/webhook \
+                 --header "Content-Type: application/json" \
+                 --data "{\"repo_name\":\"$GITHUB_REPOSITORY\",\"repo_token\":\"$GITHUB_TOKEN\",\"payload\":{\"build_num\":$GITHUB_RUN_ID,\"status\":\"done\"}}"

--- a/rebar.config.erl
+++ b/rebar.config.erl
@@ -239,12 +239,19 @@ relx_apps(ReleaseType) ->
     ++ [{N, load} || N <- relx_plugin_apps(ReleaseType)].
 
 relx_apps_per_rel(cloud) ->
-    [ {observer, load}
-    , luerl
+    [ luerl
     , xmerl
+    | [{observer, load} || is_app(observer)]
     ];
 relx_apps_per_rel(edge) ->
     [].
+
+is_app(Name) ->
+    case application:load(Name) of
+        ok -> true;
+        {error,{already_loaded, _}} -> true;
+        _ -> false
+    end.
 
 relx_plugin_apps(ReleaseType) ->
     [ emqx_retainer

--- a/rebar.config.erl
+++ b/rebar.config.erl
@@ -96,7 +96,7 @@ plugins(HasElixir) ->
 
 test_plugins() ->
     [ rebar3_proper,
-      {coveralls, {git, "https://github.com/emqx/coveralls-erl", {branch, "github"}}}
+      {coveralls, {git, "https://github.com/emqx/coveralls-erl", {branch, "fix-git-info"}}}
     ].
 
 test_deps() ->
@@ -451,7 +451,6 @@ coveralls() ->
         Cfgs = [{coveralls_repo_token, Token},
                 {coveralls_service_job_id, os:getenv("GITHUB_RUN_ID")},
                 {coveralls_commit_sha, os:getenv("GITHUB_SHA")},
-                {coveralls_service_number, os:getenv("GITHUB_RUN_NUMBER")},
                 {coveralls_coverdata, "_build/test/cover/*.coverdata"},
                 {coveralls_service_name, "github"}],
         case os:getenv("GITHUB_EVENT_NAME") =:= "pull_request"

--- a/rebar.config.erl
+++ b/rebar.config.erl
@@ -107,13 +107,13 @@ test_deps() ->
 
 common_compile_opts() ->
     [ debug_info % alwyas include debug_info
-    , deterministic
     , {compile_info, [{emqx_vsn, get_vsn()}]}
     | [{d, 'EMQX_ENTERPRISE'} || is_enterprise()]
     ].
 
 prod_compile_opts() ->
     [ compressed
+    , deterministic
     , warnings_as_errors
     | common_compile_opts()
     ].

